### PR TITLE
Add basic multiline basic LineDecoder tests

### PR DESCRIPTION
**Pull request by cosmo0920**
This PR adds basic multibyte tests to LineDecoder.
It world be better more character type in test data?

### CODE REVIEW

File: embulk-core/src/test/java/org/embulk/spi/util/TestLineDecoder.java [hunk 1]
```diff
@@ -128,5 +128,39 @@ public void testDecodeChunksCRLF() throws Exception
         assertEquals(ImmutableList.of("t1", "t2", "t3"), decoded);
     }
 
-    // TODO test multibytes
+    @Test
+    public void testDecodeBasicMultiByte() throws Exception
+    {
+        List<String> decoded = doDecode(
+                Charset.forName("utf-8"), Newline.LF,
```
Comment (frsyuki) [inline embulk-core/src/test/java/org/embulk/spi/util/TestLineDecoder.java]: Please use StandardCharsets.UTF_8 instead of Charset.forName

**Comment by frsyuki** [comment]
UTF-16 and MS932 are very good to have.

File: embulk-core/src/test/java/org/embulk/spi/util/TestLineDecoder.java (hunk 1 above)
Comment (cosmo0920) [inline embulk-core/src/test/java/org/embulk/spi/util/TestLineDecoder.java]: OK, I'll try it!

**Comment by cosmo0920** [comment]
I added following test cases:

* UTF-8 test cases which contains multibyte
* MS932 test cases which contains multibyte (for Japanese edition of Windows default encoding)
* UTF-16LE test cases which contains multibyte (this test cases for PowerShell default output text encoding)

fixes:

* Add functionality to test util function to specify charset and newline style
* Use `StandardCharsets` constants if possible

**Comment by frsyuki** [comment]
Good :+1:
I think this is good enough. But do you think you can reduce copy of code by using loop?

**Comment by cosmo0920** [comment]
I think that there is a possibility to reduce code with loop.

But this case is more elegant to use [JUnitParams](https://github.com/Pragmatists/JUnitParams) to make parametrized test.

I'll more investigate to reduce redundant test code and try to use `JUnitParams` test library.

**Comment by frsyuki** [comment]
That's interesting idea.
Or another idea is to migrate to use TestNG.
But it will be relatively large change. Let's do that using another issue thread.

**Comment by cosmo0920** [comment]
:+1: 
It sounds great.
Let's investigate above idea in another issue thread!

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**